docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,199 @@
+Z11_SPF1_MORE_THAN_ONE_SAME_MODIFIER | ERROR  | spf_modifier         | The published SPF version 1 policy contains the {spf_modifier} modifier more than once.
+Z11_SPF1_MULTIPLE_RECORDS            | ERROR   |                      | The domain publishes more than one SPF version 1 policy.
+Z11_SPF1_REDIRECT_AND_ALL            | WARNING |                      | The published SPF version 1 policy contains both the "all" keyword and a redirect modifier.
+Z11_SPF1_REDIRECT_NOT_AT_END         | WARNING |                      | The published SPF version 1 policy contains a redirect modifier, but it does not appear at the end of the string.
+Z11_SPF1_SYNTAX_ERROR                | ERROR   |                      | The published SPF version 1 policy has a syntax error.
+Z11_SPF_NOT_CONFIGURED               | INFO    |                      | The *Child Zone* does not publish an SPF policy.
+Z11_SPF_WRONG_VERSION                | WARNING | spf_version          | The *Child Zone* publishes an SPF policy with invalid version number {spf_version}.
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the term "[DNS Query]"
+follows the specification for DNS queries as specified in [DNS Query and
+Response Defaults]. The handling of the DNS responses on the DNS queries follow,
+unless otherwise specified below, what is specified for [DNS Response] in the
+same specification.
+
+
+1. Create a [DNS Query] with query type TXT query and query name *Child Zone*
+   ("TXT query").
+
+2. Create three empty sets, "TXT-Data", "SPF-Policies" and "SPFv1-Policies".
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+4. For each name server in *Name Server IP* do:
+
+   1. Send *TXT Query* to the name server and collect the [DNS Responses][DNS
+      Response].
+
+   2. Go to the next name server IP address if at least one of the following
+      criteria is met:
+
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no TXT resource record in the response.
+
+   3. If the name server responds with at least one TXT record, then, for each
+      TXT record, concatenate all strings within the TXT record’s data together
+      and add the resulting string to the *TXT-Data* set.
+
+   4. Go to the next name server.
+
+5. For each string in the *TXT-Data* set, if the string starts with `v=spf`,
+   then add it to the *SPF-Policies* set.
+
+6. If the *SPF-Policies* set is empty, then output *[Z11_SPF_NOT_CONFIGURED]*
+   and terminate the test.
```

Comment:
Do you assume that all nameservers give the same response?

Reply:
How about introducing an extra test case in the “Consistency” group which ensures all nameservers publish the same SPF policies?

Reply:
That would be OK. We could also consider saving test case and coding by including it in this test case.